### ISSUE TITLE
Update import translators for 5.0

### ISSUE THREAD

**Issue by dstillman** [zotero/translators#1353]
Import translators in 5.0 should account for the possibility of an `item.complete()` that returns a promise. This allows for items to be saved to disk individually using the asynchronous database access in 5.0. These translators should also return a promise from `doImport()`.

5.0 will assume that all import translators with a `lastUpdated` from today or later support asynchronous saving. For translators that haven't been updated, items will be added to a queue and saved together at the end of the import process, which for large imports can hang the UI during the import process.

The native Promise object is available in all modern browsers, but native promises aren't currently available in child sandboxes or in some older browsers (e.g., Safari under 10.8), so translators should use `typeof Promise == 'undefined'` to determine whether `doImport` can return a promise. (I don't think we can use a polyfill because `setTimeout()` also isn't available in the sandbox.)

For now, this won't affect web translators, including web translators that call import translators. Those will still queue items and save them at the end.

I'll post a PR shortly with an updated BibTeX translator as an example.

**Comment by twsh** [comment]
Is someone updating BibLaTeX.js? If not, I will see if I can figure it out.

**Comment by dstillman** [comment]
Yes, as long as they're not somehow called as child translators (which shouldn't be the case if you just directly triggering them with import/export processes).

**Comment by retorquere** [comment]
Where it says "the possibility of an `item.complete()` that returns a promise", can I read that as "if `async: true` is set, `item.complete()` *will* return a promise"?

Can I still import items into collections as usual?

**Comment by retorquere** [comment]
Can you walk me through what the async importers do, conceptually? I'm looking at the Zotero CSL-JSON translator as that seems the simplest translator that does async import, but I'm not fully grokking what is going on. I think that if `async` is set to `true` in the config options, `item.complete` will return a promise. Is that correct? Is `Promise.coroutine` available in the translators? Would I be mostly set by just wrapping `doImport` with `Promise.coroutine` and yielding on `item.complete()`?
